### DISCUSSION TITLE
docker compose up graphhopper

### DISCUSSION THREAD

**Discussion by papastepano** [IsraelHikingMap/Site#1708] [General]
Hello guys,
trying to execute `docker compose up graphhopper` I get an error ` java.lang.IllegalStateException: Couldn't load from existing folder: /data/default-gh/ but also cannot use file for DataReader as it wasn't specified!`
Do you have any suggestions What I am doing wrong?
BR

**Comment by HarelM** [comment]
What arguments do you use to start the docker image? Are you using a configuration file that is not the "example" one?
You need to provide more information if you want us to be able to help.

**Reply by HarelM** [reply]
Also see the instructions in the readme:
https://github.com/IsraelHikingMap/Site#setting-up-the-project-for-site-development-to-setup-ios-and-android-follow-the-cordova-guide
`docker compose up graphhopper` should fail for the first time.

**Reply by papastepano** [reply]
I am not using a different configuration, than the one provided.
For starting the docker image I use `docker compose up graphhopper` as stated here [https://github.com/IsraelHikingMap/Site#setting-up-the-project-for-site-development-to-setup-ios-and-android-follow-the-cordova-guide]. 
No matter how often I run the command I get the same error:
![image](https://user-images.githubusercontent.com/16594524/163995530-49956971-03a3-473a-87c6-90e8ebf6cec4.png)

**Reply by HarelM** [reply]
Did you run the gh-update.ps1 script?

**Reply by papastepano** [reply]
I did. It also gives an error like this:
![image](https://user-images.githubusercontent.com/16594524/164006312-cdebb1fa-c28b-4c2b-a495-8cc5c7d1182e.png)

**Reply by HarelM** [reply]
Interesting... this is probably because the manual was written before we introduced osmpbfwebapi and was not updated.
You'll need to docker compose up osmpbfwebapi and surf to its swagger to create IHM folder and serve from it the pbf file needed for graphhopper update script. you can also simply copy into the graphhopper container the osm pbf file.
Not sure if what I'm saying is cryptic or not... if it is please let me know.
If at the end you could make a PR to update the readme with your experience it would be great.
The bottom line is that you need to make the graphhopper container have data or download data in order for the routing to work...
You can also take a look at:
https://github.com/IsraelHikingMap/graphhopper-docker-image-push

**Reply by papastepano** [reply]
And where do you have this service osmpbfwebapi? I dont think its part of your docker compose yml

**Reply by HarelM** [reply]
ah, you are right, my bad.
I think the easiest solution is to change the following line:
https://github.com/IsraelHikingMap/Site/blob/e0ef172ded77a031e2ef7bde1601949a23e47f38/gh-update.ps1#L4
`osmpbfwebapi:80/IHM` --> `some-url--to-file-osm.pbf`

**Reply by papastepano** [reply]
I wanted to run this project, because you once refered to the draw implementation of maplibre. And I wanted to check it out. 
Now, to the current issue. I would not mind to contribute, but I am not familiar with the whole project. This .pbf file cannot be just any pbf file on the internet. I think it wont be straight forward for me to fix this. Maybe, when you have a chance, you should first provide this file as an asset. Cheers!

**Reply by HarelM** [reply]
If you want to do this for Israel please use the following OSM pbf url:
https://download.geofabrik.de/asia/israel-and-palestine-latest.osm.pbf

**Reply by HarelM** [reply]
In that case I would simply advise to go to israelhiking.osm.org.il and see how the editing is done, UX wise.
